### ISSUE TITLE
Perform AOT processing for autowired fields and setters

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-native#909]
`AutowiredAnnotationBeanPostProcessor` was disabled via https://github.com/spring-projects-experimental/spring-native/commit/16b62d5652ff508d8378d2671c875653d32bccd2 because it was using reflection on all the autoconfigurations which defeats the purpose of this branch in term of efficiency.

After discussion with @bclozel and @jhoeller, it seems we'd have to perform the scanning part at AOT level, and generate the code that will perform the injection at runtime.

For autowired fields and setters, we could bake similar code into the bean definition. Once we provide an instance supplier there, that supplier implementation could not only construct the object but also populate pre-detected fields/setters with `AutowireCapableBeanFactory.resolveDependency`  calls (or whatever else we do for autowired constructor arguments, it's fundamentally the same resolution algorithm).

**Comment by sdeleuze** [comment]
I tried to migrate the `webclient` sample but it fails due to no static resources serving because, as expected, [this injection](https://github.com/spring-projects/spring-framework/blob/main/spring-webflux/src/main/java/org/springframework/web/reactive/config/DelegatingWebFluxConfiguration.java#L47) does not happen.

**Comment by snicoll** [comment]
I am making some progress with method-level injection:

```java
context.registerBean("testComponent", TestComponent.class, () -> {
	TestComponent bean = new TestComponent();
	InjectionElementAttributes attributes = InjectionMethodMetadata.ofBean("testComponent", TestComponent.class)
			.forMethod("injectBean", String.class).resolve(context);
	attributes.ifResolved(() -> bean.injectBean(attributes.get(0)));
	return bean;
});
```

**Comment by snicoll** [comment]
It's worth trying the `WebSample` again once this work is merged in `0.11.x` as we generate the following now:

```java
BeanDefinitionRegistrar.of("org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration$EnableWebMvcConfiguration", WebMvcAutoConfiguration.EnableWebMvcConfiguration.class).instanceSupplier((instanceContext) -> {
  WebMvcAutoConfiguration.EnableWebMvcConfiguration bean = new WebMvcAutoConfiguration.EnableWebMvcConfiguration(context.getBean(ResourceProperties.class), context.getBean(WebMvcProperties.class), context.getBean(WebProperties.class), context.getBeanProvider(WebMvcRegistrations.class), context.getBeanProvider(WebMvcAutoConfiguration.ResourceHandlerRegistrationCustomizer.class), context);
  instanceContext.method("setConfigurers", List.class)
		  .resolve(context, false).ifResolved((attributes) -> {
	  List<WebMvcConfigurer> attributes0 = attributes.get(0);
	  bean.setConfigurers(attributes0);
  });
  return bean;
}).register(context);
```

**Comment by snicoll** [comment]
With the help of @bclozel we now have the specific reflection entries that we need to call autowired setters and fields. The `webclient` sample now fails with:

```
Caused by: java.lang.IllegalStateException: /org/springframework/http/mime.types not found in classpath
	at org.springframework.util.Assert.state(Assert.java:76) ~[na:na]
	at org.springframework.http.MediaTypeFactory.parseMimeTypes(MediaTypeFactory.java:69) ~[na:na]
	at org.springframework.http.MediaTypeFactory.<clinit>(MediaTypeFactory.java:48) ~[na:na]
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:375) ~[na:na]
	at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:295) ~[na:na]
	... 71 common frames omitted
```

**Comment by snicoll** [comment]
Closed by 0a04f4ea9033
